File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/entity/Activity.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity

import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import java.util.*
import javax.persistence.*

@Entity
class Activity(
        @Id
        @GeneratedValue
        var id: Long,
        var name: String,
        var startDate: Date,
        var endDate: Date,

        @OneToMany(mappedBy = "activity", cascade = arrayOf(CascadeType.ALL), orphanRemoval = true)
        @OnDelete(action = OnDeleteAction.CASCADE)
        private var users: List<UserActivity>? = null
)

File: spaceship_crm/scala/spaceship_crm_scala_spring_rest_jpa/src/main/scala/com/spaceship/crm/entity/Activity.scala
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity

import java.util.Date

import javax.persistence._
import org.hibernate.annotations.{OnDelete, OnDeleteAction}

import scala.beans.BeanProperty

@Entity
class Activity {

  @Id
  @GeneratedValue
  @BeanProperty var id: Long = _

  @BeanProperty var name: String = _

  @BeanProperty var startDate: Date = _
  @BeanProperty var endDate: Date = _

  @OneToMany(mappedBy = "activity", cascade = Array(CascadeType.ALL), orphanRemoval = true)
  @OnDelete(action = OnDeleteAction.CASCADE)
  var users: java.util.List[UserActivity] = new java.util.ArrayList[UserActivity]
}
